codalab/worker/file_util.py

```diff
@@ -28,19 +41,24 @@ def tar_gzip_directory(
                       the directory structure are excluded.
     ignore_file: Name of the file where exclusion patterns are read from.
     """
-    # Always ignore entries specified by the ignore file (e.g. .gitignore)
     args = ['tar', 'czf', '-', '-C', directory_path]
+
+    # If the BSD tar library is being used, prepend COPYFILE_DISABLE=1 to prevent creating ._* files
+    if 'bsdtar' in get_tar_version_output():
+        args.insert(0, 'COPYFILE_DISABLE=1')
```

Comment:
Why not just include this always?

Reply:
I originally always included it, but the commands errors with GNU tar library.

Reply:
Also, if we're already ignoreing `._*` based on `ALWAYS_IGNORE_PATTERN`, then why do we need this?

Reply:
Initially I thought just adding it to the list will work, but it seems like `._` files get generated after it excludes files by patterns.

Reply:
In that case, we shouldn't include `._*` in the `ALWAYS_IGNORE_PATTERN`, right?

Hm, running `tar` (on Linux) with `COPYFILE_DISABLE=1` works for me - it just seems to be a no-op...

Reply:
I added `._*` just in case Mac users had those files lingering around. For some reason, it errored out on the Linux machine I was using. I will try it again.